Linking error (test_atomics_executable) on Raspbian/armv6l

I am trying to compile ROS dashing from source on Raspbian Buster (`gcc (Raspbian 8.3.0-6+rpi1) 8.3.0`) running on armv6l and encounter a linking error for target `test_atomics_executable`:
```
/usr/bin/ld: CMakeFiles/test_atomics_executable.dir/test/test_atomics.c.o: in function `main':
test_atomics.c:(.text+0x11a8): undefined reference to `__atomic_store_8'
/usr/bin/ld: test_atomics.c:(.text+0x11c0): undefined reference to `__atomic_load_8'
/usr/bin/ld: test_atomics.c:(.text+0x1268): undefined reference to `__atomic_store_8'
/usr/bin/ld: test_atomics.c:(.text+0x12b8): undefined reference to `__atomic_exchange_8'
[...]
```

I am using the standard repo at https://raw.githubusercontent.com/ros2/ros2/release-latest/ros2.repos, excluding `rviz2` and `rqt`.

Edit: The problem can be resolved by manually adding `-latomic`, i.e. `target_link_libraries(test_atomics_executable ${PROJECT_NAME} -latomic)`.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,6 +11,8 @@ if(NOT CMAKE_CXX_STANDARD)
   set(CMAKE_CXX_STANDARD 14)
 endif()
 
+include(CheckLibraryExists)
+
 find_package(ament_cmake_python REQUIRED)
 find_package(ament_cmake_ros REQUIRED)
 
@@ -121,6 +123,8 @@ if(BUILD_TESTING)
 
   find_package(launch_testing_ament_cmake REQUIRED)
 
+  check_library_exists(atomic __atomic_load_8 "" HAVE_LIBATOMICS)
+
   if(ament_cmake_cppcheck_FOUND)
     ament_cppcheck(
       TESTNAME "cppcheck_logging_macros"
@@ -214,6 +218,11 @@ if(BUILD_TESTING)
       LANGUAGE C
   )
   target_link_libraries(test_atomics_executable ${PROJECT_NAME})
+  if(HAVE_LIBATOMICS)
+    target_link_libraries(test_atomics_executable atomic)
+    ament_export_libraries(atomic)
+  endif()
+
   add_test(NAME test_atomics COMMAND test_atomics_executable)
 
   rcutils_custom_add_gmock(test_error_handling test/test_error_handling.cpp

